pkg/apis/gateway/v1alpha1/types.go

```diff
@@ -0,0 +1,94 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Gateway is the definition of a gateway-controller resource
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type Gateway struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata" protobuf:"bytes,1,opt,name=metadata"`
+	Status            NodePhase   `json:"type" protobuf:"bytes,2,opt,name=type"`
```

Comment:
should this be of type `GatewayStatus` instead? make sure to edit the annotations to match the name as well!